libraries/chain/block_header_state.cpp

```diff
@@ -325,6 +380,22 @@ void finish_next(const block_header_state& prev,
       next_header_state.finalizer_policy_generation = prev.finalizer_policy_generation;
    }
 
+   // now populate next_header_state.latest_qc_claim_block_active_finalizer_policy
+   // this keeps track of the finalizer policy which was active @ latest_qc_claim().block_num, but which
+   // can be overwritten by a previously pending police (member `active_finalizer_policy`)
+   // See full explanation in issue #694.
+   // --------------------------------------------------------------------------------------------------
+   const auto& next_core                 = next_header_state.core;
+   auto        latest_qc_claim_block_num = next_core.latest_qc_claim().block_num;
+   const auto  active_generation_num     = next_header_state.active_finalizer_policy->generation;
+   if (prev.get_active_finalizer_policy_generation(latest_qc_claim_block_num) != active_generation_num) {
+      const auto& latest_qc_claim_block_ref = next_header_state.core.get_block_reference(latest_qc_claim_block_num);
+      next_header_state.latest_qc_claim_block_active_finalizer_policy =
+         prev.get_finalizer_policies(latest_qc_claim_block_ref).active_finalizer_policy;
+   } else {
+      next_header_state.latest_qc_claim_block_active_finalizer_policy = nullptr;
```

Comment:
why set to `nullptr`?

Reply:
We reset it when it doesn't need to be tracked anymore. This will allow the memory holding that policy to be release when the last block holding it as an active policy is also removed from fork_db.